src/webgpu/api/validation/encoding/cmds/index_access.spec.ts

```diff
@@ -0,0 +1,138 @@
+export const description = `
+indexed draws validation tests.
+`;
+
+import { params, poptions, pbool } from '../../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+
+import { ValidationTest } from './../../validation_test.js';
+
+class F extends ValidationTest {
+  createIndexBuffer(): GPUBuffer {
+    const indexArray = new Uint32Array([0, 1, 2, 3, 1, 2]);
+
+    const [indexBuffer, indexMapping] = this.device.createBufferMapped({
+      size: indexArray.byteLength,
+      usage: GPUBufferUsage.INDEX,
+    });
+    new Uint32Array(indexMapping).set(indexArray);
+    indexBuffer.unmap();
+
+    return indexBuffer;
+  }
+
+  createRenderPipeline(): GPURenderPipeline {
+    const vertexModule = this.makeShaderModule('vertex', {
+      glsl: `
+        #version 450
+        void main() {
+          gl_Position = vec4(0.0, 0.0, 0.0, 1.0);
+        }
+      `,
+    });
+
+    const fragmentModule = this.makeShaderModule('fragment', {
+      glsl: `
+        #version 450
+        layout(location = 0) out vec4 fragColor;
+        void main() {
+            fragColor = vec4(0.0, 1.0, 0.0, 1.0);
+        }
+      `,
+    });
+
+    return this.device.createRenderPipeline({
+      layout: this.device.createPipelineLayout({ bindGroupLayouts: [] }),
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology: 'triangle-strip',
+      colorStates: [{ format: 'rgba8unorm' }],
+    });
+  }
+
+  beginRenderPass(encoder: GPUCommandEncoder) {
+    const colorAttachment = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 1, height: 1, depth: 1 },
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    return encoder.beginRenderPass({
+      colorAttachments: [
```

Comment:
(sorry I missed this before)
nit: I'm not sure if any attachments are actually needed at all - I don't remember. Would this test be valid without them?

Reply:
The colorAttachments is required in GPURenderPassDescriptor. It will throw error if no attachment is set.

Reply:
Thank you for confirming!